components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -959,43 +962,25 @@ private static void setConsentRequiredScopesToOAuthParams(AuthenticatedUser user
             throws OAuthSystemException {
 
         try {
-            String consentRequiredScopes = StringUtils.EMPTY;
-            List<String> allowedOAuthScopes = getAllowedOAuthScopes(params);
+            //filter out OIDC scopes
+            List<String> consentRequiredScopes = dropOIDCAndUnregisteredScopesFromConsentRequiredScopes(params);
+
             if (user != null && !isPromptContainsConsent(params)) {
                 String userId = getUserIdOfAuthenticatedUser(user);
                 String appId = getAppIdFromClientId(params.getClientId());
                 OAuth2ScopeConsentResponse existingUserConsent = oAuth2ScopeService.getUserConsentForApp(
                         userId, appId, IdentityTenantUtil.getTenantId(user.getTenantDomain()));
                 if (existingUserConsent != null) {
                     if (CollectionUtils.isNotEmpty(existingUserConsent.getApprovedScopes())) {
-                        allowedOAuthScopes.removeAll(existingUserConsent.getApprovedScopes());
+                        consentRequiredScopes.removeAll(existingUserConsent.getApprovedScopes());
                     }
                 }
             }
-            if (CollectionUtils.isNotEmpty(allowedOAuthScopes)) {
-                // Filter out internal scopes to be validated.
-                String[] requestedScopes = Oauth2ScopeUtils.getRequestedScopes(
-                        allowedOAuthScopes.toArray(new String[0]));
-                if (ArrayUtils.isNotEmpty(requestedScopes)) {
-                    // Remove the filtered internal scopes from the allowedOAuthScopes list.
-                    allowedOAuthScopes.removeAll(Arrays.asList(requestedScopes));
-
-                    JDBCPermissionBasedInternalScopeValidator scopeValidator =
-                            new JDBCPermissionBasedInternalScopeValidator();
-                    String[] validatedScope = scopeValidator.validateScope(requestedScopes, user, params.getClientId());
-
-                    // Filter out requested scopes from the validated scope array.
-                    for (String scope : requestedScopes) {
-                        if (ArrayUtils.contains(validatedScope, scope)) {
-                            allowedOAuthScopes.add(scope);
-                        }
-                    }
-                }
-                params.setConsentRequiredScopes(new HashSet<>(allowedOAuthScopes));
-                consentRequiredScopes = String.join(" ", allowedOAuthScopes).trim();
-            }
+            params.setConsentRequiredScopes(new HashSet<>(consentRequiredScopes));
+            String consentRequiredScopesValue = String.join(" ", consentRequiredScopes).trim();
```

Comment:
This string is only required for the debug log. Shall we move it inside the log.isDebugEnabled() check